src/circuit/transport.ts

```diff
@@ -62,19 +80,39 @@ export class Circuit implements Transport, Startable {
     }
 
     this._started = true
+    await this.components.registrar.handle(RELAY_V1_CODEC, (data) => {
+      void this._onProtocolV1(data).catch(err => {
+        log.error(err)
+      })
+    })
+      .catch(err => {
+        log.error(err)
+      })
 
-    await this.components.registrar.handle(RELAY_CODEC, (data) => {
-      void this._onProtocol(data).catch(err => {
+    await this.components.registrar.handle(RELAY_V2_HOP_CODEC, (data) => {
```

Comment:
Should adding this listener be conditional on `HopConfig.enabled | HopConfig.active === true` ?

Reply:
Since circuit v2 kinda changed interaction, maybe it makes sense to make up new configs and names that are more dev friendly